### PR TITLE
Upload for homework 3 from Denis Zahariev

### PR DESCRIPTION
**Pull request by DeniBademi**

### CODE REVIEW

File: homework_03/08_Denis_Zahariev/README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Unit tests
```
Comment (StelianRBG) [inline homework_03/08_Denis_Zahariev/README.md:1]: I like that you uploaded a file with your assignment

File: homework_03/08_Denis_Zahariev/circle.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import numpy as np
+from enum import Enum
+
+class RelativePostion(Enum):
```
Comment (StelianRBG) [inline homework_03/08_Denis_Zahariev/circle.py:4]: good values for enum

File: homework_03/08_Denis_Zahariev/circle.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+        if self == other:
+            return RelativePostion.SAME
+
+        distance = np.sqrt((other.center.x - self.center.x) ** 2 + (other.center.y - self.center.y) ** 2) # pitagor
+
+        radius_sum = self.r + other.r
+        radius_difference = abs(self.r - other.r)
+
+        if distance == 0 and self.r == other.r:
```
Comment (StelianRBG) [inline homework_03/08_Denis_Zahariev/circle.py:46]: good idea to check for same circles

File: homework_03/08_Denis_Zahariev/test_.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
+from circle import Point, Circle, RelativePostion 
+
+# test for point initialization
+def create_point():
```
Comment (StelianRBG) [inline homework_03/08_Denis_Zahariev/test_.py:4]: to make test in start of function add test_

File: homework_03/08_Denis_Zahariev/test_.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    circle1 = Circle(Point(6.0, 6.0), 6.0)
+
+    assert circle1.find_relative_position(circle1) == RelativePostion.SAME
+
+    # test with another instance
+    circle2 = Circle(Point(6.0, 6.0), 6.0)
+
+    assert circle1.find_relative_position(circle2) == RelativePostion.SAME
+
+    circle3 = Circle(Point(6.0, 6.0), 5.0)
```
Comment (StelianRBG) [inline homework_03/08_Denis_Zahariev/test_.py:26]: this test i suggest to make in other function

File: homework_03/08_Denis_Zahariev/test_.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
+from circle import Point, Circle, RelativePostion 
+
+# test for point initialization
+def create_point():
+    point = Point(3, 2)
+    assert isinstance(point, Point)
+
+# test for circle initialization
+def create_circle() -> None:
```
Comment (StelianRBG) [inline homework_03/08_Denis_Zahariev/test_.py:9]: you dont need to add -> None it is default return value for functions

File: homework_03/08_Denis_Zahariev/test_.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    assert circle3.find_relative_position(circle4) == RelativePostion.TOUCHING
+
+def test_intersecting() -> None:
+
+    circle1 = Circle(Point(1.0, 1.0), 3.0)
+    circle2 = Circle(Point(2.0, 1.0), 3.0)
+
+    assert circle1.find_relative_position(circle2) == RelativePostion.INTERSECTING
+
+test_same()
```
Comment (StelianRBG) [inline homework_03/08_Denis_Zahariev/test_.py:66]: to test your program you dont need to call functions. pytest/covarage automaticaly run all functions start with test_

**Review by StelianRBG — APPROVED** [review]
Good work

File: homework_03/08_Denis_Zahariev/circle.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+        assert isinstance(otherCircle, Circle)
+        return otherCircle.r == self.r and otherCircle.center == self.center
+
+    def find_relative_position(self, other) -> RelativePostion:
+        assert isinstance(other, Circle)
+
+        if self == other:
+            return RelativePostion.SAME
+
+        distance = np.sqrt((other.center.x - self.center.x) ** 2 + (other.center.y - self.center.y) ** 2) # pitagor
```
Comment (Vic-Dim) [inline homework_03/08_Denis_Zahariev/circle.py:41]: Easily readable and understandable class attributes, nice

File: homework_03/08_Denis_Zahariev/circle.py [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+        radius_sum = self.r + other.r
```
Comment (Vic-Dim) [inline homework_03/08_Denis_Zahariev/circle.py:43]: This really simplifies reading the if-else clause

**Review by Vic-Dim — APPROVED** [review]

**Pull request merged**